lib/hocon/impl/tokens.rb

```diff
@@ -358,39 +390,70 @@ def self.get_substitution_optional(token)
     end
   end
 
-  def self.unquoted_text?(token)
-    token.is_a?(UnquotedText)
+  START = Token.new_without_origin(TokenType::START, "start of file", "")
+  EOF = Token.new_without_origin(TokenType::EOF, "end of file", "")
+  COMMA = Token.new_without_origin(TokenType::COMMA, "','", ",")
+  EQUALS = Token.new_without_origin(TokenType::EQUALS, "'='", "=")
+  COLON = Token.new_without_origin(TokenType::COLON, "':'", ":")
+  OPEN_CURLY = Token.new_without_origin(TokenType::OPEN_CURLY, "'{'", "{")
+  CLOSE_CURLY = Token.new_without_origin(TokenType::CLOSE_CURLY, "'}'", "}")
+  OPEN_SQUARE = Token.new_without_origin(TokenType::OPEN_SQUARE, "'['", "[")
+  CLOSE_SQUARE = Token.new_without_origin(TokenType::CLOSE_SQUARE, "']'", "]")
+  PLUS_EQUALS = Token.new_without_origin(TokenType::PLUS_EQUALS, "'+='", "+=")
+
+  def self.new_line(origin)
+    Line.new(origin)
   end
 
-  def self.unquoted_text(token)
-    if unquoted_text?(token)
-      token.value
-    else
-      raise ConfigBugOrBrokenError, "tried to get unquoted text from #{token}"
-    end
+  def self.new_problem(origin, what, message, suggest_quotes, cause)
+    Problem.new(origin, what, message, suggest_quotes, cause)
   end
 
-  def self.value?(token)
-    token.is_a?(Value)
+  def self.new_comment_double_slash(origin, text)
+    Comment::DoubleSlashComment.new(origin, text)
   end
 
-  def self.value(token)
-    if token.is_a?(Value)
-      token.value
-    else
-      raise ConfigBugOrBrokenError, "tried to get value of non-value token #{token}"
-    end
+  def self.new_comment_hash(origin, text)
+    Comment::HashComment.new(origin, text)
   end
 
-  def self.value_with_type?(t, value_type)
-    value?(t) && (value(t).value_type == value_type)
+  def self.new_unquoted_text(origin, s)
+    UnquotedText.new(origin, s)
   end
 
-  def self.newline?(t)
-    t.is_a?(Line)
+  def self.new_ignored_whitespace(origin, s)
+    IgnoredWhitespace.new(origin, s)
   end
 
-  def self.problem?(t)
-    t.is_a?(Problem)
+  def self.new_substitution(origin, optional, expression)
+    Substitution.new(origin, optional, expression)
+  end
+
+  def self.new_value(value, orig_text = nil)
+    Value.new(value, orig_text)
+  end
+
+  def self.new_string(origin, value, orig_text)
+    new_value(ConfigString.new(origin, value), orig_text)
+  end
+
+  def self.new_int(origin, value, orig_text)
+    new_value(ConfigNumber.new_number(origin, value, orig_text), orig_text)
+  end
+
+  def self.new_double(origin, value, orig_text)
```

Comment:
Hmm...it seems like a lot of instances of `float` have been changed to `double` in this code. What's the reasoning behind that?